test/unit/math/prim/scal/prob/bernoulli_test.cpp

```diff
@@ -40,3 +42,33 @@ TEST(ProbDistributionsBernoulli, chiSquareGoodnessFitTest) {
 
   EXPECT_TRUE(chi < quantile(complement(mydist, 1e-6)));
 }
+
+
+TEST(ProbDistributionsBernoulli, logitChiSquareGoodnessFitTest) {
+  boost::random::mt19937 rng;
+  int N = 10000;
+  boost::math::chi_squared mydist(1);
+ 
+  double parameter = -0.5;
+  double prob = stan::math::inv_logit(-0.5);
+  int bin[2] = {0, 0};
+  double expect [2] = {N * (1 - prob), N * prob};
+
+  int count = 0;
```

Comment:
I'd recommend this:

```
int bin[2] = { 0, 0};
for (int count = 0; count < N; ++count)
  ++bin[bernoulli_logit_rng(parameter, rng)];
```

The for loop synchs up the initialization, condition and increment to make it easier to read.  And you can just use the value of the rng directly to define the bin here.